src/adaptix/_internal/morphing/facade/provider.py

```diff
@@ -368,3 +369,7 @@ def validating_loader(data):
         raise exception_factory(data)
 
     return loader(pred, validating_loader, chain)
+
+
+def default_dict(default_factory: Optional[Type] = None) -> Provider:
+    return DefaultDictProvider(default_factory)
```

Comment:
I think that default_factory should have no default value. Builtin provider already uses ``default_factory=None``